lib/cc/cli/source_fingerprint.rb

```diff
@@ -0,0 +1,69 @@
+require "digest/md5"
+
+module CC
+  module CLI
+    class SourceFingerprint
+      def initialize(output)
+        @output = output
+      end
+
+      def compute
+        md5 = Digest::MD5.new
+        md5 << key
+        md5.hexdigest
+      end
+
+      def key
+        key = "#{path}|#{check_name}"
+        key << "|#{source.gsub(/\s+/, "")}" if source
+        key
+      end
+
+      private
+
+      attr_reader :output
+
+      def check_name
+        output["check_name"]
+      end
+
+      def path
+        output.fetch("location", {}).fetch("path", "")
+      end
+
+      def source
+        @source ||= begin
+          if line_range
+            contents = extract_source
+
+            contents unless contents.blank?
+          end
+        end
+      end
+
+      def line_range
+        @line_range ||= begin
+          lines = output.fetch("location", {})["lines"]
+
+          if lines
+            (lines["begin"]..lines["end"])
+          end
+        end
+      end
+
+      def extract_source
+        contents = ""
+
+        File.open(path) do |file|
+          file.each_line.with_index do |line, line_number|
```

Comment:
You may or may not enjoy this kind of conciseness, but I thought it was a neat trick nonetheless (to get rid of the `content` temp):

```rb
irb(main):001:0> file = "foo\nbar\nbaz\n"
irb(main):011:0> file.each_line.with_object("memo").with_index do |(l, m), ln|
irb(main):012:1* p l, m, ln
irb(main):013:1> end
"foo\n"
"memo"
0
"bar\n"
"memo"
1
"baz\n"
"memo"
2
=> "memo"
```